automated-testing/performance-testing/load-testing.md

```diff
@@ -0,0 +1,69 @@
+
+# Load Testing
+
+"*Load testing is performed to determine a system's behavior under both normal and anticipated peak load conditions.*" - [Load testing - Wikipedia](https://en.wikipedia.org/wiki/Load_testing)
+
+A load test is designed to determine how a system behaves under expected normal workload.  Specifically its main purpose is to confirm if a system can handle the expected load level. Depending on the target system this could be concurrent users, requests per second or data size.
+
+## Why Load Testing
+
+The main objective prove the system can behave normally under the expected normal load before releasing it to production. The criteria which defines "behave normally" will depend on your target, this may be as simple as "the system remains available", but it could also include meeting a response time SLA or error rate.
+
+Additionally the results of a load test can also be used as data to help with capacity planning and calculating scalability.
+
+## Load Testing Design Blocks
+
+There are a number of basic component which are required to carry out a load test.  
+
+1. In order to have meaningful results the system needs to be tested in a production-like environment with a network and hardware which closely resembles the expected deployment environment.
+
+2. The load will consist of a module which simulates user activity. Of course what the composition of this "user activity" will be vary based on the type of application being tested. For example an e-commerce web site might simulate user browsing and purchasing items, but an IoT data ingestion pipeline would simulate a stream of device readings.
+
+3. The load test will be initiated from a component external to the target system which can control the amount of load applied. This can be a single agent, but may need to scaled to multiple agents in order to achieve higher levels of activity.
+
+4. Although not required to run a load test, it is advisable to have monitoring and/or logging in place to be able to measure the impact of the test and discover potential bottlenecks.
+
+## Applying the Load Testing
+
+### Planning
+
+1. **Identify key scenarios to measure**  - Gather these scenarios from Product Owner, they should provide a representative sample of real world traffic.

```

Comment:
Part of planning could be also selected the proper load test tool. I've seen cases before where a load test tool was selected that could not generate the necessary traffic. So it might be good to indicate this is something to look at in selecting the proper tool.